app/controllers/users_controller.rb

```diff
@@ -13,11 +13,25 @@ def index
       .page(params[:page])
       .per(25)
 
-    @memberships = current_organization.members.
-      where(user_id: @users.map(&:id)).
-      includes(:account).each_with_object({}) do |mem, ob|
-        ob[mem.user_id] = mem
-      end
+    user_ids = @users.pluck(:id)
+
+    @members = current_organization.
+      members.
+      where(user_id: user_ids).
```

Comment:
can we rather do an AR merge? This avoids one SQL query which saves a network round trip, etc. PostgreSQL knows how to match against those users better than us.

Reply:
PostgreSQL knows, ActiveRecord sometimes messes up the joins. I will try to see if this is working in this case.

Reply:
Indeed it is not working in this case. As `@users` already comes from a JOIN, AR is not really able to distinguish how to prepare the association chain.

Here it is a direct ransack on members, so we don't have to mix stuff up.